### ISSUE TITLE
Questions about perspective projection

### ISSUE THREAD

**Issue by HJ-harry** [eigenvivek/DiffDRR#237]
First of all, thank you for sharing such wonderful project!

I think it's one of the most intuitive, efficient, and useful packages for X-ray/CT research. Much appreciated.

This is a question related to a closed issue, #235 but slightly different.

I'm trying to use ```drr.perspective_projection(pose, points)``` on 3D points that exist in the world coordinate to the detector plane, but I'm getting results that are flipped in an axis. The code to reproduce the problem:

```python
import matplotlib.pyplot as plt
import pyvista
import torch

from diffdrr.data import load_example_ct
from diffdrr.drr import DRR
from diffdrr.pose import convert
from diffdrr.visualization import drr_to_mesh, img_to_mesh, labelmap_to_mesh, plot_drr

import numpy as np
from pathlib import Path


device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
subject = load_example_ct()
ct = drr_to_mesh(subject, "surface_nets", threshold=225, verbose=True)

save_root = Path("./results/vis_issue")
save_root.mkdir(exist_ok=True, parents=True)
    
drr = DRR(subject, sdd=1020.0, height=200, delx=2.0).to(device)

# 3D WC points
points = torch.tensor([[[0., 0., 0.,],
                        [-100., 0., 0.]]], device=device)

zero = torch.tensor([[0.0, 0.0, 0.0]], device=device)
t = torch.tensor([[0.0, 800.0, 0.0]], device=device)
R = torch.tensor([[0.0, 0.0, 0.0]], device=device)

pose1 = convert(zero, t, parameterization="euler_angles", convention="ZXY")
pose2 = convert(R, zero, parameterization="euler_angles", convention="ZXY")
pose = pose1.compose(pose2)

pose = convert(R, t, parameterization="euler_angles", convention="ZXY")
img = drr(pose)

# visualize projection image & fiducials
points_det = drr.perspective_projection(pose, points)
axs = plot_drr(img, ticks=False)
for p in points_det.squeeze().cpu().numpy():
    axs[0].scatter(p[1], p[0], c="r", s=100)
plt.show()

camera, detector, texture, principal_ray, img = img_to_mesh(
    drr,
    pose,
)

plotter = pyvista.Plotter(off_screen=True)
plotter.add_mesh(ct)
plotter.add_mesh(camera, show_edges=True)
plotter.add_mesh(detector, texture=texture)
plotter.add_mesh(principal_ray, color="red")
plotter.add_axes()
plotter.add_bounding_box()
plotter.show_axes()
plotter.show_bounds(grid='front', location='outer', all_edges=True)

plotter.screenshot(str(save_root / f"3d_render.png"))
plotter.export_html(str(save_root / f"3d_render.html"))
plotter.close()
```
![Figure_1](https://github.com/eigenvivek/DiffDRR/assets/39315552/5b651849-db39-4625-b746-d4c57805ab1f)

![3d_render](https://github.com/eigenvivek/DiffDRR/assets/39315552/6bdb321b-a0a2-45cb-843f-6a41c22d7fc3)

Here I have two fiducials ```points_det```, where the first point is correctly marked in the center of the image plane, but the other one is placed in the left part of the image. When I try to visualize the 3d render, it's rather clear that it should actually be placed **right** of the image. Is there something that I am missing here? Thanks so much in advance :)

**Comment by eigenvivek** [comment]
ahhh i think this is an issue with the `reverse_x_axis=True` flag in the constructor for `DRR`.

if that's true, which it is by default to obey radiologic convention, then the x-axis for fiducials needs to also be flipped... ie, `drr.perspective_projection` should return `width - p[..., 1]`

thanks for catching this, i'll implement a fix

**Comment by eigenvivek** [comment]
Adding the following code to `drr.perspective_projection`

```python
    if self.detector.reverse_x_axis:
        x[..., 1] = self.detector.width - x[..., 1]
```

produces the correct image
![download](https://github.com/eigenvivek/DiffDRR/assets/29757116/21dd9346-a4e2-4e8c-9bcd-f78ade352194)
